src/libs/actions/EmojiPickerAction.ts

```diff
@@ -0,0 +1,77 @@
+import {ValueOf} from 'type-fest';
+import React from 'react';
+import {View} from 'react-native';
+import CONST from '../../CONST';
+
+type AnchorOrigin = {
+    horizontal: ValueOf<typeof CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL>;
+    vertical: ValueOf<typeof CONST.MODAL.ANCHOR_ORIGIN_VERTICAL>;
+};
+
+type EmojiPicker = {
```

Comment:
Why are we defining the component type in this file?

Reply:
It's not the component type, it's what is defined in `useImperativeHandle` [here](https://github.com/software-mansion-labs/expensify-app-fork/blob/d3b1cf5962e638007755ec2f60fc0715345a5602/src/components/EmojiPicker/EmojiPicker.js#L119). Do you think the name is misleading? @hayata-suenaga

Reply:
I mean this is component API right? the methods exposed the component through `useImperativeHandle`?

I think we should define this inside the component file instead of defining this here where we store the reference to the  component.

I know we haven't started migration of component files, but I think we should start thinking about where type definitions should belong and how to coordinates the migration when the type needed are not defined yet.

Reply:
I agree that `EmojiPicker` type should be defined next to `EmojiPicker` component. But until it's not migrated we have couple options:
- Put this PR on hold until we get to `EmojiPicker` component
- Merge this now, and move the type when `EmojiPicker` component is migrated (we can add a comment on the issue to remember about this)
- Define types.ts file with `EmojiPicker` type next to the component

Generally I think we won't be able to avoid such situations and sometimes we have to define types in undesirable places. At some point of the migration we will have to clean a bit, for example we'll have to clean style types or define shared types that are often used - until then we have to accept that we will repeat or litter the code sometimes.

cc @fabioh8010 @hayata-suenaga

Reply:
I'm confortable with option two, **and adding a TODO comment above the type as well**:

```
Merge this now, and move the type when EmojiPicker component is migrated (we can add a comment on the issue to remember about this)
```

Reply:
Maybe we will have to rename this type to e.g. `EmojiPickerRef` as well, to avoid conflicts.

Reply:
> Merge this now, and move the type when EmojiPicker component is migrated (we can add a comment on the issue to remember about this)

I also like the option #2.

> Generally I think we won't be able to avoid such situations and sometimes we have to define types in undesirable places. At some point of the migration we will have to clean a bit, for example we'll have to clean style types or define shared types that are often used

I agree. It's just we must remember all the cleanups we have to do. I don't have the capacity to manage this, so I'd appreciate if SWM and CallStack and develop a strategy to coordinate and keep track the cleanup efforts we have to do. 🙇 (like what you suggested in option 2 i.e. commenting future migration issues to remember cleanup items)

Reply:
We can create some cleanup and review issues for some of the categories of files (I kinda have been done this cleanup for `styles` already) , e.g. `styles`, `models`, `libs`, `hooks`, `hoc's`, `components` and `pages` I think are good candidates for these clean ups. After each migration of all issues of that category we will create a new issue to review all past work and ensure consistency. I can add these "checkpoints" on our spreadsheet, WDYT? @hayata-suenaga @blazejkustra

Reply:
@fabioh8010 That sounds good to me 😄 Add it to the spreadsheet please, and we will get back to this.

> I don't have the capacity to manage this, so I'd appreciate if SWM and CallStack and develop a strategy to coordinate and keep track the cleanup efforts we have to do. 🙇

Sure thing @hayata-suenaga, that's on me and @fabioh8010, we will coordinate such cleanups!

Reply:
Let me know if issues need to be created in App 🙇

Reply:
@hayata-suenaga @blazejkustra I created new rows in the spreadsheet about these review issues, here are the lines:

- 57
- 261
- 277
- 295
- 486
- 550

@hayata-suenaga Could you please create the issues? Thank you!

Reply:
@hayata-suenaga bump